lld/ELF/Arch/LoongArch.cpp

```diff
@@ -153,6 +154,20 @@ static bool isJirl(uint32_t insn) {
   return (insn & 0xfc000000) == JIRL;
 }
 
+static void handleUleb128(uint8_t *loc, uint64_t val) {
+  const uint32_t maxcount = 1 + (config->is64 ? 64 : 32) / 7;
+  uint32_t count;
+  uint64_t orig = decodeULEB128(loc, &count);
+  if (count > maxcount)
+    errorOrWarn(getErrorLocation(loc) + "extra space for uleb128");
+  uint64_t mask = config->is64 ? -1 : -1u;
```

Comment:
-1 can be used for 32-bit as well.

Reply:
If `config->is64 == false`, we expect `mask` is `0x0000'0000'ffff'ffff` so that the results will not be broken by the high-32-bits. (In 32-bits ELF file the addr size is 32bits, so the content should not be overflowed.)

Reply:
Are the relocations defined to be 32-bit on ELF64? It looks strange.

Reply:
And there is no test checking the 32-bit behavior

Reply:
I can add the same test as a.o for 32-bit.

`uleb128(-1u) == 0x0f'ff'ff'ff'ff`. Assuming it reserved 5 bytes space and we executing SUB -1 on 32-bit first, then without this mask we might get 0x7f'ff'ff'ff'ff, which [35:33] is 1. These relocations is relocated address, and address should be limited in 32-bit on ELF32. So we need this mask to avoid breaking the high-space.

Reply:
I've checked x86, mips and riscv ELF32 by GUN AS, they all handle uleb128 as 64bits value. So it's my misunderstanding. I'll correct it.